packages/camera/android/src/main/java/io/flutter/plugins/camera/CameraPlugin.java

```diff
@@ -1,88 +1,33 @@
-import android.hardware.camera2.CaptureRequest;
-import android.hardware.camera2.params.StreamConfigurationMap;
-import android.media.Image;
-import android.media.ImageReader;
-import android.media.MediaRecorder;
 import android.os.Build;
-import android.util.Size;
-import android.view.Display;
-import android.view.OrientationEventListener;
-import android.view.Surface;
 import androidx.annotation.NonNull;
-import androidx.annotation.Nullable;
 import io.flutter.plugin.common.EventChannel;
 import io.flutter.plugin.common.MethodCall;
 import io.flutter.plugin.common.MethodChannel;
 import io.flutter.plugin.common.MethodChannel.MethodCallHandler;
 import io.flutter.plugin.common.MethodChannel.Result;
-import io.flutter.plugin.common.PluginRegistry;
 import io.flutter.plugin.common.PluginRegistry.Registrar;
 import io.flutter.view.FlutterView;
-import java.io.File;
-import java.io.FileOutputStream;
-import java.io.IOException;
-import java.nio.ByteBuffer;
-import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.Collections;
-import java.util.Comparator;
-import java.util.HashMap;
-import java.util.List;
-import java.util.Map;
 
 public class CameraPlugin implements MethodCallHandler {
 
-  private static final int CAMERA_REQUEST_ID = 513469796;
-  private static final String TAG = "CameraPlugin";
-
-  private static CameraManager cameraManager;
+  private final CameraPermissions cameraPermissions = new CameraPermissions();
   private final FlutterView view;
+  private final Registrar registrar;
+  private final EventChannel imageStreamChannel;
   private Camera camera;
-  private Registrar registrar;
-  // The code to run after requesting camera permissions.
-  private Runnable cameraPermissionContinuation;
-  private final OrientationEventListener orientationEventListener;
-  private int currentOrientation = ORIENTATION_UNKNOWN;
 
-  private CameraPlugin(Registrar registrar, FlutterView view) {
+  private CameraPlugin(Registrar registrar) {
     this.registrar = registrar;
-    this.view = view;
-
-    orientationEventListener =
-        new OrientationEventListener(registrar.activity().getApplicationContext()) {
-          @Override
-          public void onOrientationChanged(int i) {
-            if (i == ORIENTATION_UNKNOWN) {
-              return;
-            }
-            // Convert the raw deg angle to the nearest multiple of 90.
-            currentOrientation = (int) Math.round(i / 90.0) * 90;
-          }
-        };
-
-    registrar.addRequestPermissionsResultListener(new CameraRequestPermissionsListener());
+    this.view = registrar.view();
+    this.imageStreamChannel =
+        new EventChannel(registrar.messenger(), "plugins.flutter.io/camera/imageStream");
   }
 
   public static void registerWith(Registrar registrar) {
-    if (registrar.activity() == null || Build.VERSION.SDK_INT < Build.VERSION_CODES.LOLLIPOP) {
+    if (Build.VERSION.SDK_INT < Build.VERSION_CODES.LOLLIPOP) {
```

Comment:
Did you mean to remove this? I believe this check is needed when you are embedding the plugin in an Android app.

Reply:
Yes, I meant to remove it. In general, it is a bad idea to *not* register plugins. Activity is technically not needed when this plugin is registered, so why require it?

We should probably remove the version check as well. Often apps (Dart) do not check the presence of activity or SDK version before calling a plugin. To make it as user friendly as possible, plugins should degrade gracefully and return proper errors.

If you don't register the plugin altogether, you get a catastrophic failure that says "channel name not found" which is not indicative of the real problem.

Reply:
The version check was added in this [PR](https://github.com/flutter/plugins/pull/1598) and was only added so a developer could still compile the plugin with sdk < 21. To do this, one would need to add a specific line in their manifest to ignore that this plugin requires 21+.